src/openrct2/interface/Screenshot.cpp

```diff
@@ -550,15 +550,9 @@ int32_t cmdline_for_gfxbench(const char** argv, int32_t argc)
 
 static void ApplyOptions(const ScreenshotOptions* options, rct_viewport& viewport)
 {
-    if (options->weather != 0)
+    if (options->weather != WeatherType::Sunny && options->weather != WeatherType::Count)
     {
-        if (options->weather < 1 || options->weather > 6)
-        {
-            throw std::runtime_error("Weather can only be set to an integer value from 1 till 6.");
-        }
-
-        uint8_t customWeather = options->weather - 1;
-        climate_force_weather(customWeather);
+        climate_force_weather(weatherTypes[EnumValue(options->weather) - 1]);
```

Comment:
Not sure this `weatherTypes` solution is the best, why not just use it directly? You've changed `ScreenshotOptions` to use a WeatherTYpe already, so there shouldn't be any conversion.

Reply:
Thank you for the great feedback! I used the weatherTypes array here because I first needed the enum value to be able to subtract one from it. 

I kept the subtraction but to be fair, it seems like either a bug or a hack to me.

When you say I can use it directly, does that imply I am allowed to subtract 1 from an enumeration and then cast it back to an enum?

Reply:
I hadn't noticed the subtraction, so you might be right, though I guess it's better to subtract and cast it back (?)

Reply:
This function is already performing all the validation to make sure that doing -1 on the value will produce the correct result. I'd prefer if you just used EnumValue directly.

Reply:
I have changed it now :) I used the underlying enumvalue, subtracted one and then passed it to the constructor (?) if that's what this is called: `WeatherType{ }`